src/utils/ProtectedRoutes.tsx

```diff
@@ -0,0 +1,17 @@
+import { apiRootStore } from '@/store/apiRootStore';
+import { Navigate, Outlet } from 'react-router';
+
+type ProtectedRoutesProps = {
+  requiredLoginState: boolean;
+  redirectedPath?: string;
+};
+
+export const ProtectedRoutes = ({
+  requiredLoginState,
+  redirectedPath = '/',
+}: ProtectedRoutesProps) => {
+  if (apiRootStore().isLoggedIn !== requiredLoginState) {
```

Comment:
```suggestion
   const { isLoggedIn } = useApiRootStore();
   if (isLoggedIn !== requiredLoginState) {
```

Use imported hook

Now if we call `store.login` for example in loginPage, `store.isLoggedIn` will become true and `ProtectedRoutes` rerenders and Navigate will work

Now it rerenders Component and redirects to `/`:
<img width="1886" alt="image" src="https://github.com/user-attachments/assets/82789b30-9e34-4641-b1e3-7f2eb8f28bcf" />

Reply:
I rewrote this moment, now auto redirect is working.